### ISSUE TITLE
Arrow key in spell target mode casts the spell adjacent to you

### ISSUE THREAD

**Issue by NQNStudios** [calref/cboe#482]
So if you're casting Fireball.... you've just pressed one button (probably thinking it would move the camera) and possibly killed your caster.

**Comment by CelticMinstrel** [comment]
Using the arrow key for targeting is intended behaviour though. I suppose it's unfortunate that you can accidentally hit yourself when doing so, but I'm not really convinced it counts as a bug.

**Comment by NQNStudios** [comment]
Yeah, well I’ve observed that there are multiple contexts where you’re in a
targeting mode and in some of them you’d want the arrow keys to target next
to you, while in others you’d want the camera to move. Sometimes it would
be impossible for us to programmatically tell the difference, i.e. in look
mode it’s very convenient to target next to you for searching things, but
the player could just as easily be using look mode to see farther away, and
they’d want camera movement instead. When targeting ranged, you’d almost
never want to target adjacent to you.

Maybe the main point here is that there’s *no* keyboard input way to move
the camera while targeting.

**Comment by NQNStudios** [comment]
I made an issue on my fork's tracker that's for quality-of-life improvements, some of which might change the experience so you might not want them in the upstream (I'll still PR and you can decide individually). Here it is: https://github.com/NQNStudios/cboe/issues/16

If you consider this issue to fall into that category, we can close it here and move it onto my QOL checklist.

**Comment by CelticMinstrel** [comment]
I don't see anything unreasonable on that list, so feel free to open each of those as an issue over here if you want. I wouldn't say I'm on-board with everything on that list, but I could probably be convinced for most of them. The one that seems the most dubious is skill training everywhere…

Back on-topic, I can see your point about not having a way to move the camera with the keyboard. That's also true in look mode, though. I also see your point that it would be very unusual to want to cast a ranged spell, especially an area-of-effect spell, on an adjacent target. But actually changing how this works would probably be quite difficult, as the arrow keys are simply mapped to be equivalent to a keypress on the adjacent space.

So, I'm not sure what's the best thing to do here.

**Comment by NQNStudios** [comment]
Are there any Mage/Priest spells that can only target an adjacent space? I think if the arrow key shortcut doesn't cover every meaningful input in the spellcasting context, we should just turn it off for spellcast targeting.

**Comment by CelticMinstrel** [comment]
Good examples of spells where you'd _want_ to use an arrow key for targeting are unlock or move mountains. Though those are cast in town generally. (Not sure if they're available in combat.)
